### ISSUE TITLE
GraphQL sync endpoint: `KeyError` always assumed to be an input problem

### ISSUE THREAD

**Issue by mroach** [rmosolgo/graphql-ruby#4699]
**Describe the bug**

We ran into an issue when syncing the schema and we got `422 Unprocessable Entity` as a response code with `"Invalid input structure"` as the response body. This was really confusing considering the input was perfectly valid and took a fair bit of work and time to debug.

I ended up having to manually copy the generated JSON body into a script and run the operation sync manually to find the actual source of the problem. Something like this:

```ruby
begin
  GraphQL::Pro::OperationStore::AddOperationBatch.call(
    operation_store: FrontendSchema.operation_store,
    client_name: "frontend",
    operations:,
    changeset_version:
  )
rescue KeyError => err
  puts "KEY ERROR WITH #{err.key}"
  puts err.backtrace
end
```

The problem as I see it is that `graphql/pro/operation_store/endpoint.rb` has a `rescue KeyError` in the context of the entire HTTP request which means *any* KeyError is caught and returned to the client with "Invalid input structure".

In our specific case here, we had code doing `context.fetch(:current_user)` assuming there would be a user, there wasn't, so boom.

What would be a lot more helpful is if this error were only rescued for the most specific context where it makes sense, which is probably just `payload.fetch("operations")`. If an error is raised while actually running the operation store routine, I'd want that error to be raised so it can be logged like a normal exception and the client gets a 500.

**Versions**

`graphql`: 2.1.5
`graphql-pro`: 1.24.9
`graphql-enterprise`: 1.3.1
`rails`: 7.0.8

**Comment by rmosolgo** [comment]
Hey, sorry for the trouble with this and thanks for the detailed report. I definitely agree that more specific error handling would be a big improvement here -- I'll take a look and follow up back here soon!

**Comment by rmosolgo** [comment]
Thanks again for reporting this. I just released GraphQL-Pro v1.24.13 which moves the `rescue KeyError`s right next to the `.fetch` calls, so they won't interfere with other error bubbling. Please give that version a try and let me know if you run into any trouble with it!

**Comment by mroach** [comment]
@rmosolgo Thanks so much! We fixed our underlying bug but if we do something similar again, it'll be nice to have this. Thanks :)
